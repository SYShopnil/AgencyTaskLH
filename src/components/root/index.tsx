import Button from './button';

export { Button };
export { BannerSectionBar } from './banner-bar';
export { SlideCard } from './slide-card/';
export { HeroSectionWithOnScrollRotation } from './hero-section-with-on-scroll-rotaition';
export { Navbar } from './navbar';
