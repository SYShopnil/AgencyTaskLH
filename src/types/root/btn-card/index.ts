export enum ColorSchema {
  PrimaryYellowBgTextWhite = 'PrimaryYellowBgTextWhite',
  PrimaryBlueBgTextWhite = 'PrimaryBlueBgTextWhite',
  SecondaryLightGreenBgTextWhite = 'SecondaryLightGreenBgTextWhite',
  SecondaryLightVioletBgTextWhite = 'SecondaryLightVioletBgTextWhite',
  SecondaryOrangeBgTextWhite = 'SecondaryOrangeBgTextWhite',
  SecondaryPinkBgTextWhite = 'SecondaryPinkBgTextWhite',
  SecondaryBlueBgTextWhite = 'SecondaryBlueBgTextWhite',
  SecondaryDeepGreenBgTextWhite = 'SecondaryDeepGreenBgTextWhite',
  AccentGrayBgTextWhite = 'AccentGrayBgTextWhite',
  AccentBlackBgTextWhite = 'AccentBlackBgTextWhite',
  LightBlueBgTextWhite = 'LightBlueBgTextWhite',
}

export enum CardSize {
  card_sm = 'card_sm',
  card_lg = 'card_lg',
}
export interface IBtnCard {
  title: string;
  colorSchema: ColorSchema;
  cardSize: CardSize;
  fill?: number;
  data_test_id?: string;
}
