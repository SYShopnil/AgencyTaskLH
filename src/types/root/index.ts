export type { IButton, INormalButton, IRedirectButton } from './Button/';

export { BtnColorSchema } from './Button/';

export type { IBtnCard } from './btn-card';
export { ColorSchema, CardSize } from './btn-card';

export type { IIconStore } from './_icon';
export { IconName as IconEnums } from './_icon';

export type { ICurveFrame } from './curveFrame';
export { CurveFrameVariantEnum } from './curveFrame';

export type { IInfoCard } from './infoCard';
export type { ITitleDescSection } from './title_desc_sec/';

export type { IBanner, IBannerSectionBar } from './banner-bar/banner.ts';
export type { ISlideCard } from './slide-card';

export type { IHeroSectionWithOnScrollRotation } from './hero-section-with-on-scroll-rotaition/index';
export type { IFooter } from './footer';

export type {
  ICommonSection,
  IMegaMenu,
  IMegaMenuItems,
  ISubMenuBlock,
  INavbar,
} from './navbar';
