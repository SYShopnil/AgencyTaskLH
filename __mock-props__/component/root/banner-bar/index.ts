import { IBanner, IBannerSectionBar } from '@src/types/root';
import { IconName } from '@src/types/root/_icon';

export const BannerImageMockData: IBanner[] = [
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
];

export const BannerBarSectionMockData: IBannerSectionBar = {
  contents: BannerImageMockData,
  bannerBarSectionTestId: 'mainWrapper',
};
