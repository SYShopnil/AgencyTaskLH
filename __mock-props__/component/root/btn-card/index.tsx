import { CardSize, ColorSchema, IBtnCard } from '@src/types/root';

export const btnCardMockProps: IBtnCard[] = [
  {
    colorSchema: ColorSchema.PrimaryBlueBgTextWhite,
    title: 'Grass',
    cardSize: CardSize.card_lg,
    fill: 65,
    data_test_id: 'btnCardOne',
  },
  {
    colorSchema: ColorSchema.PrimaryBlueBgTextWhite,
    title: 'Grass',
    cardSize: CardSize.card_sm,
    fill: 50,
    data_test_id: 'btnCardTwo',
  },
];
