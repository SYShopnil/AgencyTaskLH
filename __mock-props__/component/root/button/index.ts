import { BtnColorSchema, IRedirectButton } from '@src/types/root';

export const mock_data_redirectButton: IRedirectButton[] = [
  {
    btnText: 'Back to Homepage',
    colorSchema: BtnColorSchema.BlackBgTextWhite,
    isOpenNewTab: true,
    redirectLink: 'https://chat.openai.com/',
    dataTestId: 'Redirect Button One',
  },
  {
    btnText: 'Back to Homepage Two',
    colorSchema: BtnColorSchema.BlackBgTextWhite,
    isOpenNewTab: false,
    redirectLink: '',
    dataTestId: 'Redirect Button Two',
  },
];
