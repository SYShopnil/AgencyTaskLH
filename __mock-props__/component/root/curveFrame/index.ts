import { CurveFrameVariantEnum, ICurveFrame } from '@src/types/root';

export const mockPropsOfCurveFrame: ICurveFrame[] = [
  {
    variant: CurveFrameVariantEnum.with_gradient,
    children: `<div}>
                <p>Hello world</p>
                <p>Hello world</p>
                <p>Hello world</p>
                <p>Hello world</p>
            </div>`,
    dataTestIdForCurveFrame: 'frameCurveOne',
    borderWidth: '2%',
  },
  {
    variant: CurveFrameVariantEnum.with_out_gradient,
    children: `<div}>
                <p>Hello world</p>
                <p>Hello world</p>
                <p>Hello world</p>
                <p>Hello world</p>
            </div>`,
    dataTestIdForCurveFrame: 'frameCurveTwo',
    borderWidth: '2%',
  },
];
