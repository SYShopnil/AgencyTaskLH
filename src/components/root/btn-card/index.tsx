import { CardSize, ColorSchema, IBtnCard } from '@src/types/root';
import React from 'react';

const BtnCard = ({
  colorSchema,
  title,
  cardSize,
  fill,
  data_test_id,
}: IBtnCard) => {
  let cardSizeStyle = 'px-[1.25rem] py-[0.625rem] text-md';
  switch (cardSize) {
    case CardSize.card_lg: {
      cardSizeStyle = `px-[1.25rem] py-[0.625rem] text-md`;
      break;
    }
    case CardSize.card_sm: {
      cardSizeStyle = `px-[1.25rem] text-base`;
      break;
    }
    default: {
      cardSizeStyle = 'px-[1.25rem] py-[0.625rem] text-md';
    }
  }
  const dynamicStyle = {
    top: 0,
    left: 0,
    background: 'red',
    height: `${fill}%`,
    width: '100%',
    zIndex: -1,
  };

  //control the btn card color dynamically
  switch (colorSchema) {
    case ColorSchema.AccentBlackBgTextWhite: {
      dynamicStyle.background = '#000000';
      break;
    }
    case ColorSchema.PrimaryBlueBgTextWhite: {
      dynamicStyle.background = '#2B73B9';
      break;
    }
    case ColorSchema.SecondaryBlueBgTextWhite: {
      dynamicStyle.background = '#4592C4';
      break;
    }
    case ColorSchema.SecondaryLightGreenBgTextWhite: {
      dynamicStyle.background = '#9BCC50';
      break;
    }
    case ColorSchema.SecondaryOrangeBgTextWhite: {
      dynamicStyle.background = '#FC7C23';
      break;
    }
    case ColorSchema.AccentGrayBgTextWhite: {
      dynamicStyle.background = '#D9D9D9';
      break;
    }
    case ColorSchema.PrimaryYellowBgTextWhite: {
      dynamicStyle.background = '#FFCB05';
      break;
    }
    case ColorSchema.SecondaryLightVioletBgTextWhite: {
      dynamicStyle.background = '#B97FC9';
      break;
    }
    case ColorSchema.SecondaryDeepGreenBgTextWhite: {
      dynamicStyle.background = '#729F3F';
      break;
    }
    case ColorSchema.SecondaryPinkBgTextWhite: {
      dynamicStyle.background = '#F366B9';
      break;
    }
    case ColorSchema.LightBlueBgTextWhite: {
      dynamicStyle.background = '#3DC7EF';
      break;
    }
    default: {
      dynamicStyle.background = '#4592C4';
    }
  }
  return (
    <div className="z-0" data-testid={data_test_id ? data_test_id : 'btn-card'}>
      <span
        className={`z-[-5] relative duration-150 inline-block bg-[#BDBDBD] text-center ${cardSizeStyle}  text-white rounded-[0.25rem] !leading-normal`}
      >
        <span
          className={`absolute rounded-[0.25rem]`}
          style={dynamicStyle}
          data-testid="dynamicStyleTestId"
        ></span>
        <span className={`z-10`}>{title}</span>
      </span>
    </div>
  );
};

export default BtnCard;
