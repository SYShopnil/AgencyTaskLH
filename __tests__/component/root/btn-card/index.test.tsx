import { btnCardMockProps } from '@root/__mock-props__/component/root/btn-card';
import BtnCard from '@src/components/root/btn-card';
import { render } from '@testing-library/react';

let btnCount = 0; //keep track the btn card's serial

/**
 *  DONE
 *
 * Take Snapshot
 * Check Fill works or not
 *
 *
 */

/**
 * TO DO
 *
 *
 *
 */

describe.each(btnCardMockProps)(
  'When all button card variants has been rendered',
  (card) => {
    btnCount++;
    // test start
    it(`Expect Btn card ${btnCount} Rendered Successfully`, () => {
      const { getAllByTestId } = render(<BtnCard {...card} />);
      const btnCardElements = getAllByTestId(card.data_test_id || 'btn-card');
      expect(btnCardElements).toMatchSnapshot();
    });
    it(`Expect Fill of Btn card ${btnCount}'s fill functionality worked perfectly`, () => {
      const { getByTestId } = render(<BtnCard {...card} />);
      const dynamicStyleWrapperSpanInHtml = getByTestId('dynamicStyleTestId');
      const appliedHeight =
        dynamicStyleWrapperSpanInHtml.style.height.split('%')[0];
      expect(card.fill?.toString()).toEqual(appliedHeight);
    });
  }
);
