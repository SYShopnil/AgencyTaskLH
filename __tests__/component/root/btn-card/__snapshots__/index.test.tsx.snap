// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When all button card variants has been rendered Expect Btn card 1 Rendered Successfully 1`] = `
[
  <div
    class="z-0"
    data-testid="btnCardOne"
  >
    <span
      class="z-[-5] relative duration-150 inline-block bg-[#BDBDBD] text-center px-[1.25rem] py-[0.625rem] text-md  text-white rounded-[0.25rem] !leading-normal"
    >
      <span
        class="absolute rounded-[0.25rem]"
        data-testid="dynamicStyleTestId"
        style="top: 0px; left: 0px; background: rgb(43, 115, 185); height: 65%; width: 100%; z-index: -1;"
      />
      <span
        class="z-10"
      >
        Grass
      </span>
    </span>
  </div>,
]
`;

exports[`When all button card variants has been rendered Expect Btn card 2 Rendered Successfully 1`] = `
[
  <div
    class="z-0"
    data-testid="btnCardTwo"
  >
    <span
      class="z-[-5] relative duration-150 inline-block bg-[#BDBDBD] text-center px-[1.25rem] text-base  text-white rounded-[0.25rem] !leading-normal"
    >
      <span
        class="absolute rounded-[0.25rem]"
        data-testid="dynamicStyleTestId"
        style="top: 0px; left: 0px; background: rgb(43, 115, 185); height: 50%; width: 100%; z-index: -1;"
      />
      <span
        class="z-10"
      >
        Grass
      </span>
    </span>
  </div>,
]
`;
